src/vs/base/common/async.ts

```diff
@@ -19,18 +19,19 @@ export interface CancelablePromise<T> extends Promise<T> {
 }
 
 export function createCancelablePromise<T>(callback: (token: CancellationToken) => Promise<T>): CancelablePromise<T> {
-	const source = new CancellationTokenSource();
+	const disposableStore = new DisposableStore();
+	const source = disposableStore.add(new CancellationTokenSource());
```

Comment:
Not sure why the source is wrapped in a dispo store? Is this to capture the listener from onCancellationRequested? This shouldn't be a problem when the store is always disposed. Btw, this made me realize that there is still a leak here: when cancellation happens (L26/27) the source/disposableStore isn't disposed

Reply:
Thanks for pointing out the leak!

> Is this to capture the listener from onCancellationRequested?

Yes.

> This shouldn't be a problem when the store is always disposed.

You are right, it does not need to be disposed here as the disposables inside of `CancellationTokenSource` are not tracked.
It is very important to always dispose the event subscription though, otherwise the tooling introduced by this PR will recognize that as leak (even though garbage collection does take care of it).